### ISSUE TITLE
How can I run the video module to achieve real-time detection?

### ISSUE THREAD

**Issue by ll1214** [bonlime/keras-deeplab-v3-plus#25]
Hello I want to know the speed of deeplabv3+ ,and I try to run that:
    from keras.preprocessing.image import ImageDataGenerator, array_to_img, img_to_array, load_img
from matplotlib import pyplot as plt
import cv2 # used for resize. if you dont have it, use anything else
import numpy as np
from model import Deeplabv3
deeplab_model = Deeplabv3()
def detect_video(deeplab_model):
    import cv2
    vid = cv2.VideoCapture(0)
    if not vid.isOpened():
        raise IOError("Couldn't open webcam or video")
    accum_time = 10
    curr_fps = 10
    fps = "20"
    prev_time = timer()
    while True:
        return_value, frame = vid.read()
        res = deeplab_model.predict(frame)
		
		result = array_to_img(res)
        curr_time = timer()
        exec_time = curr_time - prev_time
        prev_time = curr_time
        accum_time = accum_time + exec_time
        curr_fps = curr_fps + 1
        if accum_time > 1:
            accum_time = accum_time - 1
            fps = "FPS: " + str(curr_fps)
            curr_fps = 0
        cv2.putText(result, text=fps, org=(3, 15), fontFace=cv2.FONT_HERSHEY_SIMPLEX,
                    fontScale=0.50, color=(255, 0, 0), thickness=2)
        cv2.namedWindow("result", cv2.WINDOW_NORMAL)
        cv2.imshow("result", result)
        if cv2.waitKey(1) & 0xFF == ord('q'):
            break
    deeplab_model.close_session()
detect_video(deeplab_model)

but it doesn't work!
I thank I need your help thanks.

**Comment by bonlime** [comment]
If you want to get useful help, you need to make everything as clear as you can. 
I can't help you, because "It doesn't work" can mean nearly everything.

**Comment by ll1214** [comment]
Thank you for your warm reply. I'm sorry, there's no specific information. I read your test code, knowing that I first read a picture and scale the length and width of the image. Finally, the prediction function is used. So I changed it on your code, changed it to a camera to catch a picture, predict the picture, and finally show the prediction in a window. However, the result of window detection is not the result of prediction, but the capture of the camera in real time, and there is no corresponding file in the folder.
And my computer's GPU was not working at that time. I don't know what's the reason.

**Comment by ll1214** [comment]
And I want to ask how fast deeplabv3+ real-time detection speed, I want to use yolov3 and deeplabv3+ for real-time detection to achieve the effect of instance detection, do you think it is feasible?

**Comment by bonlime** [comment]
@ll1214  i'm not sure it can work real-time due to big size, you will need some hacks to make it work

**Comment by vodp** [comment]
@ll1214 from my experiments, deeplabv3+ with mobilenetv2 running on K80, batchsize=2 imagesize 512x512, achieves ~0.5-0.6 sec per frame.

**Comment by ll1214** [comment]
@bonlimeYes, this is because of the size of the picture, I just solved it, but my problem now is that I can't display the predicted results in OpenCV. The picture is 21 dimensional.  and my code is 
      ''  resized2 = np.pad(resized,((0,pad_x),(0,0),(0,0)),mode='constant')
        
        res = deeplab_model.predict(np.expand_dims(resized2,0))
        
        labels = np.argmax(res.squeeze(),-1)
        cv2.namedWindow("result", cv2.WINDOW_NORMAL)
        cv2.imshow("result", labels[:-pad_x])
''
and have this error:
   
 cv2.imshow("result", labels[:-pad_x])

error: C:\projects\opencv-python\opencv\modules\highgui\src\window.cpp:331: error: (-215) size.width>0 && size.height>0 in function cv::imshow
   I hope you can help me.Thank you very much!
